test/basic-rollback-test.js

```diff
@@ -16,7 +16,10 @@ test('rolling back transaction calls ROLLBACK', assert => {
   domain1.run(() => {
     return db.transaction(() => {
       throw new Error('no thanks')
-    })().reflect()
+    })().then(
```

Comment:
im looking at the reflect docs http://bluebirdjs.com/docs/api/reflect.html

it returns a custom object, but we dont use that object. 

`null` seems very specific here. 

i'd rather this be .catch(() => undefined) which clearly communicates to swallow the error